docs/installation/sources/sources_linux.rst

```diff
@@ -0,0 +1,210 @@
+
+Finally, use colcon_ to compile all software:
+
+.. code-block:: bash
+
+    colcon build
+
+Once that’s finished building, you can source the new colcon overlay:
+
+.. code-block:: bash
+
+    source install/setup.bash
+
+.. important::
+
+    The sourcing of the local colcon overlay is required every time the colcon workspace is opened in a new shell
+    environment to run an *eProsima Fast DDS* instance.
+    As an alternative, you can add it permanently to you path by typing the following:
+
+    .. code-block:: bash
+
+        echo 'source PATH_TO_WORKSPACE/Fast-DDS-ws/install/setup.bash' >> ~/.bashrc
+
+    Where :code:`PATH_TO_WORKSPACE` is the path to the :code:`Fast-DDS-ws` worskspace.
+
+.. note::
+
+    If you want to compile the examples, you will need to add the flag
+    :code:`--cmake-args "-DCOMPILE_EXAMPLES=ON"` when running :code:`colcon build`.
+    If you want to compile the performance tests, you will need to add the flag
+    :code:`--cmake-args "--DPERFORMANCE_TESTS=ON"` when running :code:`colcon build`.
+    For this step, you need Gtest_ as explained in the requirements_linux_sources_ section above.
+
+
+.. _cmake_installation_linux:
+
+CMake Installation
+------------------
+
+This section explains how to compile *eProsima Fast DDS* locally with CMake_.
+First of all, create a Fast-DDS directory where to download and build *eProsima Fast DDS* and its dependencies:
+
+.. code-block:: bash
+
+    mkdir Fast-DDS && cd Fast-DDS
+
+Now clone the following dependencies and compile them using CMake_.
+
+* `Foonathan memory <https://github.com/foonathan/memory>`_
+
+  .. code-block:: bash
+
+      $ git clone https://github.com/eProsima/foonathan_memory_vendor.git
+      $ mkdir foonathan_memory_vendor/build && cd foonathan_memory_vendor/build
+      $ cmake .. -DCMAKE_INSTALL_PREFIX=../../install
+      $ sudo cmake --build . --target install
+      $ cd ../..
+
+* `Fast CDR <https://github.com/eProsima/Fast-CDR.git>`_
+
+  .. code-block:: bash
+
+      $ git clone https://github.com/eProsima/Fast-CDR.git
+      $ mkdir Fast-CDR/build && cd Fast-CDR/build
+      $ cmake .. -DCMAKE_INSTALL_PREFIX=../../install
+      $ sudo cmake --build . --target install
+      $ cd ../..
+
+Once all dependencies are installed, you will be able to compile and install *eProsima Fast DDS*:
+
+.. code-block:: bash
+
+    $ git clone https://github.com/eProsima/Fast-RTPS.git
+    $ mkdir Fast-RTPS/build && cd Fast-RTPS/build
+    $ cmake ..  -DCMAKE_INSTALL_PREFIX=../../install -DCMAKE_PREFIX_PATH=../../install
+    $ sudo cmake --build . --target install
+
+
+
+If you want to install *eProsima Fast DDS* system-wide instead of locally, you need to remove all the flags that
+appear in the configuration steps of :code:`Fast-CDR` and :code:`Fast-RTPS`, and change the one in the
+configuration step of :code:`foonathan_memory_vendor` to the following:
+
+.. code-block:: bash
+
+    -DCMAKE_INSTALL_PREFIX=/usr/local/ -DBUILD_SHARED_LIBS=ON
```

Comment:
I'd actually put `-DBUILD_SHARED_LIBS=ON` in the local installation as well.

Reply:
only for foonathan?

Reply:
yes. adjust text of system-wide part accordingly.

Reply:
ok.